### DISCUSSION TITLE
Help with java auto instrumentation in GKE

### DISCUSSION THREAD

**Discussion by sreejesh-radhakrishnan-db** [open-telemetry/opentelemetry-java-instrumentation#8526] [Q&A]
Hi
My limited knowledge on this subject  hence trying my luck here.

I am on GKE 

I am following the https://opentelemetry.io/docs/k8s-operator/automatic/#create-an-opentelemetry-collector-optional , I could see Operator, Collector up and running in my cluster and I annotated the container too. Still no trace/metric/log is been pushed to collector log nor on application container log or to cloud trace.

I can see the INIT CONTAINER IS starting up and there is only one command " cp /javaagent.jar /otel-auto-instrumentation/javaagent.jar"

My Q - should I have to explicitly change the java command line for the application to use this jar? if so how can I do that?

Should I add in my deployment something like this
env:
- name: JAVA_TOOL_OPTIONS
value: /otel-auto-instrumentation/javaagent.jar
If we have to start up agent, then how can we do it? as the INIT container is copying the file to a location which is ROOT and we only allow NONROOT users in our deployment. 

I was advised this should be taken care of if in K8s,  but my Q is if its taken care of, how/who is enabling the agent in my POD and how it data exported?   (https://github.com/open-telemetry/opentelemetry-operator/issues/1744)   any help would be much appreciated.

**Comment by trask** [comment]
hi @sreejesh-radhakrishnan-db! can you try using the OpenTelemetry Java agent without using the OpenTelemetry Operator? that will help narrow down whether your issue is with the Java agent (this repo) or with the Operator (other repo)

**Comment by sreejesh-radhakrishnan-db** [comment]
@trask   please find the finding below


To add this also resulted in same error:

A ) I added agent to my image and tried to start  (used 1.26 - opentelemetry-javaagent.jar downloaded from the release page added to my base image)  
B) I added image with CHMOD, CHCHAGEOWNER and tried ( COPY --chown=nonroot:nonroot --from=build --chmod=765 ./javaagent.jar ${WORK_DIR}/javaagent.jar

this time as well I got same error as below


Picked up JAVA_TOOL_OPTIONS: -javaagent:/app/javaagent.jar
Error opening zip file or JAR manifest missing : /app/javaagent.jar
Error occurred during initialization of VM



Note: I am using alpine for base JRE image -  does optl jar has any libc dependency at all?

**Reply by laurit** [reply]
@sreejesh-radhakrishnan-db That error usually means that `/app/javaagent.jar` doesn't exist. Check that the agent jar exists at the that path and that it is readable.

**Reply by sreejesh-radhakrishnan-db** [reply]
Hi, yes it exist and I have done an ls -ltr on the POD, rest assured the jar does exit. 

/app:
-rwxrw-r-x 1 nonroot nonroot  javaagent.jar

**Reply by laurit** [reply]
If it exists then verify that the jar hasn't been damaged, check the size or checksum. If that looks ok try to run `java -javaagent:/path/to/javaagent.jar -version` if that still fails with the same error then try moving the jar to a different location.
As this does not seem to be a javaagent issue there isn't much we can do to help you.

**Reply by sreejesh-radhakrishnan-db** [reply]
sure. But to get , does anyone run the baseimage created of alpine where I was told libc support is not there.
